### ISSUE TITLE
Incorrect disconnect reason

### ISSUE THREAD

**Issue by DmitryBk** [doghappy/socket.io-client-csharp#182]
Hi,
It seems for `WebSocketMessageType.Close` the lib should use real disconnected reason received from the server and send "io client disconnect" reason to the server for explicit disconnection.
PR #181

**Comment by doghappy** [comment]
Judging from the signature of CloseAsync, the second string parameter is a description of the first parameter. Setting its value to "io client disconnect" is only friendly to humans, and it seems that the socket.io server does not use it.

Socket.io defines some disconnect reasons: https://socket.io/docs/v4/client-api/#Event-%E2%80%98disconnect%E2%80%99

---

You modified DisconnectAsync(), but if no exception is thrown, the `OnDisconnected` event will not be triggered

**Comment by DmitryBk** [comment]
socket.io server disconnects us and sends this reason ("io client disconnect") back to us. The client will receive `WebSocketMessageType.Close` message with this reason and calls `OnClosed(result.CloseStatusDescription ?? string.Empty)` that invokes `OnDisconnected` event.

**Comment by DmitryBk** [comment]
yep, currently with my PR `OnDisconnected` event really is not invoked for this case because in `DisconnectAsync` we have code that manually set `Connected = false; Disconnected = true;`. I think we move this code to catch block also, and these values will be changed in `InvokeDisconnectAsync`

**Comment by doghappy** [comment]
> I think we should remove this code, and these values will be changed in `InvokeDisconnectAsync`

exactly

After the socket.io server is disconnected, CloseStatusDescription is empty.

![image](https://user-images.githubusercontent.com/20038914/127980937-f37e6ea1-e4ed-4289-a795-67d14aadcc8c.png)

**Comment by DmitryBk** [comment]
Currently it is empty, but if we send our description in 
`_ws.CloseAsync(WebSocketCloseStatus.NormalClosure, "io client disconnect", CancellationToken.None);` , it will have our description
![image](https://user-images.githubusercontent.com/18037978/127982392-6a286b7e-6dc6-4cf7-bb6c-684f5d13e740.png)

**Comment by doghappy** [comment]
Still got empty, socket.io server: v4 

![image](https://user-images.githubusercontent.com/20038914/127983317-4024237b-8945-4b30-b836-597365544eaf.png)

**Comment by DmitryBk** [comment]
Strange. I`ve checked only with my v3 socket.io server and it works fine for me

**Comment by doghappy** [comment]
Still got empty, socket.io server: v3 🎃

**Comment by DmitryBk** [comment]
It seems not all servers return statusDescription back. Not sure but I think it`s server side bug. Can we make something like this?
![image](https://user-images.githubusercontent.com/18037978/127985311-bd26d3bb-c08c-419c-8192-07356d9eec39.png)

**Comment by doghappy** [comment]
No, the `CloseStatusDescription` is unknown, in order to keep the same with the socket.io documentation, we cannot change the reason yet.

I pushed a [commit](https://github.com/doghappy/socket.io-client-csharp/commit/6b62ba68140153f7de691f1393e26e1bc6399b4b), changed the order of OnClosed calls, and temporarily solved it.
